lib/malloc.h

```diff
@@ -20,6 +20,58 @@
 #include "defines.h"
 
 
+#define CALLOC(n, type)        ((type *) calloc(n, sizeof(type)))
+#define XCALLOC(n, type)       ((type *) xcalloc(n, sizeof(type)))
+#define MALLOCARRAY(n, type)   ((type *) mallocarray(n, sizeof(type)))
+#define XMALLOCARRAY(n, type)  ((type *) xmallocarray(n, sizeof(type)))
+
+#define ALLOCA(type)           ALLOCARRAY(1, type)
+#define MALLOC(type)           MALLOCARRAY(1, type)
+#define XMALLOC(type)          XMALLOCARRAY(1, type)
+#define REALLOC(ptr, type)     REALLOCARRAY(ptr, 1, type)
+#define REALLOCF(ptr, type)    REALLOCARRAYF(ptr, 1, type)
+
+#define REALLOCARRAY(ptr, n, type)                                            \
+({                                                                            \
+	__auto_type  p_ = (ptr);                                              \
```

Comment:
Is `__auto_type` standard C?

Reply:
No.  It's GNU C, and available in GCC and Clang.  ISO C23 will likely add `auto` with the same meaning, AFAIK (and hopefully deprecate the old meaning of `auto`).

Reply:
BTW, another GNU extension used here (also available at least in GCC and Clang) is `({ ... })`: <https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html>